### DISCUSSION TITLE
如何使用ilogtail收集k8s监控信息和事件信息？

### DISCUSSION THREAD

**Discussion by simops-328** [alibaba/ilogtail#314] [Help] [answered]
我们希望使用ilogtail收集kubernetes管控组件 exporter ，以此来获取 Kubernetes 管控组件的观测数据；通过 cAdvisor 获取容器的资源观测数据；通过 kube-state-metrics 获取 Kubernetes 资源的状态数据，还有事件和 Kubernetes 资源的状态以及条件数据。
有没有一个很好的建议

**Comment by messixukejia** [comment, marked as answer]
## K8s事件
k8s事件是存在ETCD里，通过API Server暴露的，iLogtail没有提供直接的采集K8s事件的插件。SLS商业版是基于NPD+kube eventer来做的，直接往SLS写。详见：https://help.aliyun.com/document_detail/125679.htm?spm=a2c4g.11186623.0.0.56576d01sPF7Uu#section-rya-7i0-9e3
![image](https://user-images.githubusercontent.com/13250519/184614651-67fc07bb-2939-4635-b842-0e5051713035.png)

也可以使用https://github.com/opsgenie/kubernetes-event-exporter将事件转成stdout，在用ilogtail采集容器的标准输出完成。

ilogtail也会评估下对接apiserver直接采集的能力。

## k8s监控信息
可以使用service_prometheus插件来进行采集Prometheus exporter 数据，可以参考下https://developer.aliyun.com/article/871634#slide-1
